snuba/web/db_query.py

```diff
@@ -34,12 +35,25 @@
     RateLimitExceeded,
 )
 from snuba.util import force_bytes, with_span
-from snuba.utils.codecs import JSONCodec, JSONData
+from snuba.utils.codecs import Codec
 from snuba.utils.metrics.timer import Timer
 from snuba.web import QueryException, QueryResult
 
-cache: Cache[JSONData] = RedisCache(
-    redis_client, "snuba-query-cache:", JSONCodec(), ThreadPoolExecutor()
+
+class ResultCacheCodec(Codec[bytes, Result]):
+    def encode(self, value: Result) -> bytes:
+        return cast(str, rapidjson.dumps(value)).encode("utf-8")
+
+    def decode(self, value: bytes) -> Result:
+        ret = rapidjson.loads(value)
+        assert (
```

Comment:
Is there a chance we would actually hit this case? If so, maybe an AssertionError isn't the correct exception?

Reply:
It would happen if we have corrupted data in cache. This is why I added the message. I can turn it into valueerror. What do you think ?

Reply:
That seems better. Will this bubble up as a cache miss in that case?

Reply:
It would fail in a similar way as it fails today. Later in the pipeline somebody tries to use the content and they would throw there if we do not throw here.